sandbox/update-local-registry.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python3
+
+import enum
+import csv
+import subprocess
+import argparse
+import tempfile
+import shutil
+import os
+import logging
+from typing import List, Set, Dict, Iterable, Optional
+
+import katsdpcontroller.generator
+
+
+class Action(enum.Enum):
+    COPY = 1
+    BUILD = 2
+    TUNE = 3
+
+
+class ImageInfo:
+    def __init__(self, *,
+                 action: Action = Action.COPY,
+                 repo: Optional[str] = None,
+                 branch: str = 'master'):
+        self.action = action
+        self.repo = repo
+        self.branch = branch
+
+
+# Order is important here: images are built in order
+EXTRA_IMAGES = ['docker-base-runtime', 'docker-base-build',
+                'docker-base-gpu-build', 'docker-base-gpu-runtime',
+                'katsdpingest', 'katsdpcontroller']
+IMAGES = EXTRA_IMAGES + list(katsdpcontroller.generator.IMAGES)
+
+IMAGE_INFO = {
+    'docker-base-runtime': ImageInfo(repo='katsdpdockerbase'),
+    'docker-base-build': ImageInfo(repo='katsdpdockerbase'),
+    'docker-base-gpu-build': ImageInfo(action=Action.BUILD, repo='katsdpdockerbase'),
+    'docker-base-gpu-runtime': ImageInfo(action=Action.BUILD, repo='katsdpdockerbase'),
+    'katsdpcal': ImageInfo(action=Action.BUILD, repo='katsdppipelines'),
```

Comment:
Note that this isn't a complete list - it's just images that can't be handled in a "default" way. I'm hoping that it will disappear over time, particularly if we manage to resolve the CUDA licensing issues.